esp/clients/ws_dfsclient/ws_dfsclient.cpp

```diff
@@ -316,70 +317,74 @@ class CServiceDistributedFile : public CServiceDistributedFileBase<IDistributedF
     CServiceDistributedFile(IDFSFile *_dfsFile) : PARENT(_dfsFile)
     {
         IPropertyTree *file = dfsFile->queryFileMeta()->queryPropTree("File");
-        const char *remotePlaneName = file->queryProp("@group");
-        VStringBuffer planeXPath("planes[@name=\"%s\"]", remotePlaneName);
-        IPropertyTree *filePlane = dfsFile->queryCommonMeta()->queryPropTree(planeXPath);
-        assertex(filePlane);
-        const char *remoteName = dfsFile->queryRemoteName(); // NB: null if local
 
+        const char *remoteName = dfsFile->queryRemoteName(); // NB: null if local
         if (!isEmptyString(remoteName))
         {
-            // Path translation is necessary, because the local plane will not necessarily have the same
-            // prefix. In particular, both a local and remote plane may want to use the same prefix/mount.
-            // So, the local plane will be defined with a unique prefix locally.
-            // Files backed by URL's or hostGroups will be access directly, are not mounted, and do not require
-            // this translation.
-            const char *filePlanePrefix = filePlane->queryProp("@prefix");
-            if (isAbsolutePath(filePlanePrefix) && !filePlane->hasProp("@hosts")) // otherwise assume url
+            Owned<IPropertyTree> remoteStorage = getRemoteStorage(remoteName);
+            if (!remoteStorage)
+                throw makeStringExceptionV(0, "Remote storage '%s' not found", remoteName);
+            if (!remoteStorage->getPropBool("@useDafilesrv"))
```

Comment:
NB: the change in this function is mostly block movement. The significant change is the addition of this check.
@useDafileSrv can be defined as a property of the remote definition, meaning it will force a remap to the direct dafilesrv service, and avoid plane remapping.


It is intended for future use where bare-metal systems would connect/pull data off a dafilesrv not via mounts, but used during testing, reading files via daliadmin remotetest.